apis/apiextensions/v1/composition_types.go

```diff
@@ -265,6 +276,64 @@ type PatchPolicy struct {
 	FromFieldPath *FromFieldPathPolicy `json:"fromFieldPath,omitempty"`
 }
 
+// ConstantType is the type of constant
+type ConstantType string
+
+// Accepted ConstantTypes
+const (
+	ConstantTypeBool   ConstantType = "bool"
+	ConstantTypeString ConstantType = "string"
+	ConstantTypeInt    ConstantType = "int"
+)
+
+// A ConstantValue is constant value to patch
+type ConstantValue struct {
+	// Type of the Constant, defaults to string.
+	// +kubebuilder:validation:Enum=bool;int;string
```

Comment:
I recently learned that you can associate this particular validation with the type declaration. I have a slight preference for keeping it there since it's perhaps a little closer to where the allowed values are defined.

Reply:
I'm a little confused with this one, did you mean to reference the Enum or the bool/int/string constants, where I am using a kubebuilder marker for the type.

```go
// Bool is a constant boolean value
	// +kubebuilder:validation:Type=boolean
	// +optional
	Bool *bool `json:"bool,omitempty"`
```

Assuming it is the latter, I will remove the unnecessary marker.

Reply:
markers for bool and int removed in b440ebc9

Reply:
I was referring to the enum - apologies for the vaguery. What I was aiming to communicate was that this is an option too:

```go
// ConstantType is the type of constant
// +kubebuilder:validation:Enum=bool;int;string
type ConstantType string
```

Very much a nitpick, but I do think there's a little value in keeping the validation closer to where the allowed values are defined.